bitnami/nginx/values.yaml

```diff
@@ -198,6 +196,130 @@ service:
   ##
   externalTrafficPolicy: Cluster
 
+
+## LDAP Auth Daemon Properties
+##
+## Daemon that will proxy LDAP requests
+## between NGINX and a given LDAP Server
+##
+ldapDaemon:
+
+  enabled: false
+
+  ## Bitnami NGINX LDAP Auth Daemon image
+  ## ref: https://hub.docker.com/r/bitnami/nginx-ldap-auth-daemon/tags/
+  ##
+  image:
+    registry: docker.io
+    repository: bitnami/nginx-ldap-auth-daemon
+    tag: 0.20200116.0-debian-10-r7
+    pullPolicy: IfNotPresent
+
+  ## LDAP Daemon port
+  ##
+  port: 8888
+
+  ## LDAP Auth Daemon Configuration
+  ##
+  ## These different properties define the form of requests performed
+  ## against the given LDAP server
+  ##
+  ## BEWARE THAT THESE VALUES WILL BE IGNORED IF A CUSTOM LDAP SERVER BLOCK
+  ## ALREADY SPECIFIES THEM.
+  ##
+  ##
+  ldapConfig:
+
+    ## LDAP URI where to query the server
+    ## Must follow the pattern -> ldap[s]:/<hostname>:<port>
+    uri: ""
```

Comment:
There is something that I don't understand, we are giving the possibility of providing LDAP config from values like `ldapDeamon.ldapConfig.uri` but after that, we should include de config again in `nginxServerBlock` ? Actually, we could build this `nginxServerBlock` file automatically in the secret, right? and provide just an only one entry point for configuration. If the user wants to provide a custom `nginxServerBlock` it should use `existingNginxServerBlockSecret`, shouldn't he?

Also, I don't understand why we pass these `ldapDeamon.ldapConfig...` through env vars to the container and also we pass the secret with the service block.
- we should ensure a mechanism of avoiding passing env vars if the service block exists and vice versa.
- the service block by default should be built in the secrets automatically.

Reply:
Actually, the user is given three possibilities to configure this LDAP Daemon:

1. Providing values using `ldapConfig` (and thus env vars)
2. Providing an `nginxServerBlock` that explicitly declare these values, in which case, will overwrite whatever value `ldapConfig` specify.
3. Providing the name of a Secret that holds this configuration `existingNginxServerBlockSecret`.

In any of the cases, a server_block that tells NGINX to use the proxy (and which are the routes that should remain confidential) is needed. Thus, we can't automatically build the file; and even if we could it should be based on the `ldapConfig` properties. They are independent.